.grit/patterns/java/correct_true_comparison_operator.md

```diff
@@ -0,0 +1,34 @@
+---
+title: Correct comparison operator `$x = true`
+tags: [good-practice]
+---
+
+Update redundant incorrect comparison `($x == true)` to achieve clearer code logic and avoid unnecessary repetition.
+
+```grit
+language java
+
+`if($var = true) { $body }` => `if($var == true) { $body }`
```

Comment:
Scoping the top level rewrite more is usually better as the diagnostic is more specific.

```suggestion
`$var = true` => `$var == true` where {
  $var <: within `if ($cond) { $body }`,
  $var <: within $cond,
}
```